src/writer.rs

```diff
@@ -470,6 +470,35 @@ impl WriterBuilder {
         self
     }
 
+    /// The comment character that will be used when later reading the file.
+    ///
+    /// If `quote_style` is set to `QuoteStyle::Necessary`, a field will
+    /// be quoted if the comment character is detected anywhere in the field.
+    ///
+    /// The default value is None.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::error::Error;
+    /// use csv::WriterBuilder;
+    ///
+    /// # fn main() { example().unwrap(); }
+    /// fn example() -> Result<(), Box<dyn Error>> {
+    ///     let mut wtr =
+    ///         WriterBuilder::new().comment(Some(b'#')).from_writer(Vec::new());
+    ///     wtr.write_record(&["# comment", "another"]).unwrap();
+    ///     let buf = wtr.into_inner().unwrap();
+    ///     assert_eq!(String::from_utf8(buf).unwrap(), "\"# comment\",another\n");
```

Comment:
This runs with `cargo test` (`cargo test --doc comment` to run just this one), so my understanding of the earlier review is that comment_char_is_automatically_quoted can be removed

Reply:
I'm aware, but @BurntSushi asked me to add both a doctest and a test at the bottom of the file.

Reply:
That's a different file, for `csv-core`. The test at the bottom of this file should be removed because it is mostly superfluous to the doc test. The other test is for `csv-core`, which exists at a lower level of abstraction and is useful to test independently.

Reply:
Ah, that makes more sense - fixed.